teos/api.py

```diff
@@ -239,31 +250,44 @@ def get_appointment(self):
         locator = request_data.get("locator")
 
         try:
-            with self.rw_lock.gen_rlock():
-                self.inspector.check_locator(locator)
-                self.logger.info(
-                    "Received get_appointment request", from_addr="{}".format(remote_addr), locator=locator
-                )
-                appointment_data, status = self.watcher.get_appointment(locator, request_data.get("signature"))
+            self.inspector.check_locator(locator)
+            self.logger.info("Received get_appointment request", from_addr="{}".format(remote_addr), locator=locator)
 
-            if status == "being_watched":
-                # Cast the ExtendedAppointment to Appointment to remove all the tower-specific data
-                appointment_data = Appointment.from_dict(appointment_data).to_dict()
-            else:
-                # Remove user_id field from appointment data since it is an internal field.
-                appointment_data.pop("user_id")
+            with grpc.insecure_channel(f"{self.internal_rpc_host}:{self.internal_rpc_port}") as channel:
+                stub = HTTP_APIStub(channel)
+                r = stub.get_appointment(
+                    GetAppointmentRequest(locator=locator, signature=request_data.get("signature"))
+                )
+                data = r.appointment_data.appointment if r.status == "being_watched" else r.appointment_data.tracker
 
-            rcode = HTTP_OK
-            response = {"locator": locator, "status": status, "appointment": appointment_data}
+                rcode = HTTP_OK
+                response = {
+                    "locator": locator,
+                    "status": r.status,
+                    "appointment": json_format.MessageToDict(
+                        data, including_default_value_fields=True, preserving_proto_field_name=True
+                    ),
+                }
 
-        except (InspectionFailed, AuthenticationFailure, AppointmentNotFound):
+        except (InspectionFailed, grpc.RpcError):
             rcode = HTTP_NOT_FOUND
             response = {"locator": locator, "status": "not_found"}
 
         return jsonify(response), rcode
 
-    def start(self):
-        """ This function starts the Flask server used to run the API """
 
-        # ToDo: #185-serve-teosd-production
-        self.app.run(host=self.host, port=self.port)
+def serve(btc_rpc_user, btc_rpc_password, btc_rpc_connect, btc_rpc_port, min_to_self_delay, log_file):
+    setup_logging(log_file)
+    inspector = Inspector(
+        BlockProcessor(
+            {
+                "BTC_RPC_USER": btc_rpc_user,
+                "BTC_RPC_PASSWORD": btc_rpc_password,
+                "BTC_RPC_CONNECT": btc_rpc_connect,
+                "BTC_RPC_PORT": int(btc_rpc_port),
+            }
+        ),
+        int(min_to_self_delay),
+    )
+    api = API(inspector)
+    return api.app
```

Comment:
serve is an outside method from the API class. Since now the API is run externally (via gunicorn atm) it needs a method to hook to. serve is basically creating the API object and returning the app object so gunicorn can run it.